### ISSUE TITLE
help! getting "Error processing request." when I click authorize...(both for anilist and mal)

### ISSUE THREAD

**Issue by RashikFarhan** [vosmiic/jellyfin-ani-sync#90]
/* Please do not create an issue about a specific show or movie not working with the plugin. Shows that have multiple cours or odd seasons generally do not work well with this plugin, and we do not fix individual issues. */

**Describe the bug**
Everything goes perfectly until the test auth. where it produces a auth link for mal or anilist. when I open it it asks for permission by clicking Allow. But when I click allow, an error message occurs "Error processing request.". It's happening for both MAL and Ani;ist.
**To Reproduce**
Follow the instructions of the plugin and make api and generate auth link.
open the link, click Allow
it'll give "Error processing request."

**Desktop (please complete the following information):**
 - Browser Chrome
 - Jellyfin version 10.8.12
 - Plugin version 2.9

**Comment by vosmiic** [comment]
Please provide your Jellyfin logs

**Comment by RashikFarhan** [comment]
Sorry, the log file is too big...cleanup logs isn't working...I hope you can find the required infos
[Log.txt](https://github.com/vosmiic/jellyfin-ani-sync/files/13401139/Log.txt)

**Comment by vosmiic** [comment]
What redirect URL are you using? From your logs (you may want to edit/delete the attached file, as it may contain some information you do not want to keep public) I can tell it should probably be `http://localhost:8096/AniSync/apiUrlTest`
Did you also test the redirect URL to confirm it is correct?

**Comment by sean-mcgauley** [comment]
Hello just commenting to say that I have run into the same issue as above. Sometimes the link brings up a login prompt and fails with the message "invalid client" when cancel is hit; I managed to get to the above mentioned screen once as well without having to log in where you would hit allow on MAL but then it returns "error processing request".  The ApiUrlTest passes.

I believe this would be the relevant error found in the jellyfin logs:
[ERR] [36] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request: "Could not retrieve Mal token: Unauthorized - Unauthorized". URL "GET" "/AniSync/authCallback".

As an aside I noticed the link that gets generated when hitting "authorize user" might have a typo?
towards the end it's redirect_uri which gives me the login prompt but fails; if I change it to redirect_url then I get to the page with "allow" which also still fails with "error processing request".  The status code I have seen in various places is "400"

https://myanimelist.net/v1/oauth2/authorize?response_type=code&client_id=6beda3b3035f923d4b66191bdfd02814&code_challenge=eZBLUX_JPk4~el62z_k3Q4fV5CzCYHoTz4iLKvwJ~9QTsTJNlzwveKCSYCSiSOa5zAm5Zt~cfyVM~3BuO4kQ0iYwCxPoeN0SOmBYR_C.QgnzyYE4KY-xIe4Vy1bf7_B4&redirect_uri=XXX.XXX.X.XX:8096/AniSync/authCallback
